src/functional/iterator/embedded.py

```diff
@@ -586,27 +594,31 @@ def shape(self):
         return self.array().shape
 
 
+def _is_tuple_axis(axis: Axis):
```

Comment:
```suggestion
def _is_tuple_axis(axis: Axis) -> bool:
```
Additionally, this typing doesn't make sense. Either `Axis` is an `Union` including `None` or the type of the axis parameter is different here.  In case `Axis` should be an `Union`, this function could be implemented differently as a TypeGuard:

```
FieldAxis: TypeAlias = Dimension
Axis: TypeAlias = FieldAxis: | None

...

def _is_field_axis(obj) -> TypeGuard[FieldAxis]:
    return isinstance(obj, Axis)
```

Reply:
I implemented your suggestion